lib/autoupdate/tools/updater.go

```diff
@@ -237,26 +250,26 @@ func (u *Updater) Update(ctx context.Context, toolsVersion string) error {
 		return trace.Wrap(err)
 	}
 
+	var pkgNames []string
 	for _, pkg := range packages {
-		if err := u.update(ctx, pkg); err != nil {
+		pkgName := fmt.Sprint(uuid.New().String(), updatePackageSuffix)
+		if err := u.update(ctx, pkg, pkgName); err != nil {
 			return trace.Wrap(err)
 		}
+		pkgNames = append(pkgNames, pkgName)
+	}
```

Comment:
The uuid is here to have a temporary package location unique to the process? Then we copy the content and remove it?

Later, we do `os.CreateTemp(u.toolsDir)`, putting the temporary package in the user's home rather than the system's default `tmp`? Is there any advantage/issue leading us to use the user's home as a temporary directory versus te default temp location?

Reply:
we should store original binary in same directory, since this is not temporary files
```
~/.tsh/bin/tctl    symlynk->    ~/.tsh/bin/UUID/tctl
~/.tsh/bin/tsh     symlynk->    ~/.tsh/bin/UUID/tsh
```
rather than
```
~/.tsh/bin/tctl    symlynk->    /tmp/Random/tctl
```

Reply:
I think the issue is with the archive, not the extracted location:
https://github.com/gravitational/teleport/pull/47815/files#diff-6ddd22bd400fc52c2062de3b1fc775fdaf05594f85c5ffa5a02262af40f5d62dR275

In general, it's unusual to use `os.CreateTemp` outside of `/tmp`.